openlibrary/book_providers.py

```diff
@@ -577,6 +581,24 @@ def get_cover_url(ed_or_solr: Edition | dict) -> str | None:
     return None
 
 
+def get_dimensions_cover(url: str) -> tuple[int, int] | None:
```

Comment:
Since we have access to the database that contains all the cover info, instead of making a network request, it would be faster to instead make a database request.

I think calling the `details` method should fix this:

https://github.com/internetarchive/openlibrary/blob/8b2a94044f01b9388b96ff094559c8e4637b6d4f/openlibrary/coverstore/db.py